src/librustc/ty/layout.rs

```diff
@@ -356,33 +372,37 @@ impl Integer {
             (I16, false) => tcx.types.u16,
             (I32, false) => tcx.types.u32,
             (I64, false) => tcx.types.u64,
+            (I128, false) => tcx.types.u128,
             (I1, true) => tcx.types.i8,
             (I8, true) => tcx.types.i8,
             (I16, true) => tcx.types.i16,
             (I32, true) => tcx.types.i32,
             (I64, true) => tcx.types.i64,
+            (I128, true) => tcx.types.i128,
         }
     }
 
     /// Find the smallest Integer type which can represent the signed value.
     pub fn fit_signed(x: i64) -> Integer {
```

Comment:
These currently do not take i128, because this PR does not adapt the compiler to support enum discriminants with 128-bit sized values. There’s a FIXME comment a few lines below along the lines of “this should take i128?”.